cf/data/data.py

```diff
@@ -10078,67 +10074,52 @@ def flatten(self, axes=None, inplace=False):
           [15 19 23]]]
 
         """
-        if inplace:
-            d = self
-        else:
-            d = self.copy()
+        d = _inplace_enabled_define_and_cleanup(self)
 
-        ndim = self._ndim
+        ndim = d.ndim
         if not ndim:
             if axes or axes == 0:
                 raise ValueError(
-                    "Can't flatten: Can't remove an axis from "
-                    "scalar {}".format(self.__class__.__name__)
+                    "Can't flatten: Can't remove an axes from "
```

Comment:
(Feels a bit pedantic to mention, but since you changed it from one to the other here) surely the original, `axis`, is in fact the right one?

Reply:
Not at all. Fixed as: `"Can't flatten: Can't remove axes from "`